### ISSUE TITLE
Single manager can define resolvers with the same url

### ISSUE THREAD

**Issue by paul121** [regen-network/regen-ledger#1212]
## Summary of Bug

https://github.com/regen-network/regen-ledger/pull/1128 removed the unique constraint on URLs across all managers. This makes it possible for multiple managers to define a resolver with the same URL.

As a result, right now it is possible for a single manager to define multiple resolvers with the same url themselves. Since each resolver still gets a unique ID and anchored data is registered to a resolver using the unique ID, this isn't really a technical issue. However, I don't think this is necessarily a "feature" either.

What does it mean if a manager defines two resolvers with the same URL? This can happen quite easily if the same `regen tx define-resolver` command is sent twice. On mainnet if someone registers the same URL twice, which resulting ID should they use? It seems that there should be a unique constraint on URLs for a single manager.

## Version

https://github.com/regen-network/regen-ledger/commit/f21333c248971df0aaf1b6b56f9bea9901bd3dde

## Steps to Reproduce

Define two resolvers with the same URL from the same manager account:

```
$ regen-ledger/build/regen q data resolver 1
resolver:
  id: "1"
  manager: regen129cwzpkph3kk2fwqc3jvugcf4wfllwf9yyredu
  url: http://localhost:8000/resource
$ regen-ledger/build/regen q data resolver 2
resolver:
  id: "2"
  manager: regen129cwzpkph3kk2fwqc3jvugcf4wfllwf9yyredu
  url: http://localhost:8000/resource
```

Also observe that the same data can be registered to both of these resolvers (requires a `RegisterResolver` message for each). Again, not a technical issue, but what does this mean?

```
$ regen-ledger/build/regen q data resolvers-by-iri regen:13toVgf5aZqSVSeJQv562xkkeoe3rr3bJWZgg1jJfnzPGBtJNpVVakm.rdf
pagination:
  next_key: null
  total: "2"
resolvers:
- id: "1"
  manager: regen129cwzpkph3kk2fwqc3jvugcf4wfllwf9yyredu
  url: http://localhost:8000/resource
- id: "2"
  manager: regen129cwzpkph3kk2fwqc3jvugcf4wfllwf9yyredu
  url: http://localhost:8000/resource
```

____

## For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by ryanchristo** [comment]
> It seems that there should be a unique constraint on URLs for a single manager.

Yea, seems like this is simply a user error that could be prevented.

> On mainnet if someone registers the same URL twice, which resulting ID should they use?

In this situation, I guess the manager would choose one to maintain and ignore the other.

> Also observe that the same data can be registered to both of these resolvers (requires a RegisterResolver message for each). Again, not a technical issue, but what does this mean?

Regardless of who the manager is (same or otherwise), two resolvers with the same URL that have registered the same data would just mean two resolvers are claiming to serve the same data. If iterating through resolvers claiming to serve the same data, the data would be fetched after the first attempt so there be no need to fetch to attempt to fetch it a second time.

**Comment by ryanchristo** [comment]
> It seems that there should be a unique constraint on URLs for a single manager.

Yea, seems like this is simply a user error that could be prevented.

> On mainnet if someone registers the same URL twice, which resulting ID should they use?

In this situation, I guess the manager would choose one to maintain and ignore the other.

> Also observe that the same data can be registered to both of these resolvers (requires a RegisterResolver message for each). Again, not a technical issue, but what does this mean?

Regardless of who the manager is (same or otherwise), two resolvers with the same URL that have registered the same data would just mean two resolvers are claiming to serve the same data. If iterating through resolvers claiming to serve the same data, and the data was successfully fetched after the first attempt, there be no need to fetch it a second time.

**Comment by paul121** [comment]
> Regardless of who the manager is (same or otherwise), two resolvers with the same URL that have registered the same data would just mean two resolvers are claiming to serve the same data. If iterating through resolvers claiming to serve the same data, and the data was successfully fetched after the first attempt, there be no need to fetch it a second time.

This is true, it is effectively the same situation for anyone wanting to consume anchored data from the resolvers. Will close this then as it likely won't create any other issues, just remain an oddity :-)

**Comment by ryanchristo** [comment]
I think the unique constraints on resolver url and manager would still be nice and not too much trouble.

**Comment by ryanchristo** [comment]
Might be nice to prevent the user error. Can be fixed with #1219
